modulemd/v2/modulemd-component.c

```diff
@@ -48,6 +48,18 @@ enum
 static GParamSpec *properties[N_PROPS];
 
 
+gboolean
+modulemd_component_equals_wrapper (const void *a, const void *b)
+{
+  g_return_val_if_fail (MODULEMD_IS_COMPONENT ((ModulemdComponent *)a), FALSE);
+  g_return_val_if_fail (MODULEMD_IS_COMPONENT ((ModulemdComponent *)b), FALSE);
+
+  return modulemd_component_equals (
+    MODULEMD_COMPONENT ((ModulemdComponent *)a),
```

Comment:
You don't need to wrap this in `MODULEMD_COMPONENT()` because you've already validated it with `MODULEMD_IS_COMPONENT()` above. So just cast it directly (needed to avoid the `const` warnings) and skip the extra cast.

```suggestion
    (ModulemdComponent *)a,
```